documentation/pdb_support_guide.markdown

```diff
@@ -31,15 +31,78 @@ The terminus resides on the Puppet master and redirects agent data to
 PDB in the form of "commands". PDB has four commands, as described in the
 [commands documentation][commands].
 
-### Queue
+### Message Queue
 
-The commands sent from the terminus to PDB are likely to be deferred
-to a filesystem queue for later processing, at which point they will
-be submitted to the database.  Commands may not always be processed in
-the order received, but will ususally be handled in roughly the order
-submitted.  Although "stale" commands might be ignored.  For example,
-older fact sets or catalogs that are still in the queue might be
-dropped when newer ones are received for the same certname.
+The commands sent from the terminus to PDB are likely to be (currently
+always) deferred to a filesystem queue for later processing, at which
+point they will be submitted to the database.  Commands may not always
+be processed in the order received, but will ususally be handled in
+roughly the order submitted, though "stale" commands might be ignored.
+For example, older fact sets or catalogs that are still in the queue
+might be dropped when newer ones are received for the same certname.
+
+Previous versions of PDB stored deferred messages in ActiveMQ, which
+conglomerated them in opaque binary files inside `vardir`/mq.  Now PDB
+stores the messages via [stockpile], which records each message as a
+normal file inside `vardir`/stockpile/cmd/q.  Whenever PDB starts up
+it will check to see if it has already migrated older messages from
+ActiveMQ to stockpile, and if not, it will migrate them, and make a
+note not to attempt the migration again.
+
+The message names are designed to provide some information about their
+content.  For example:
+
+    stockpile/cmd/q/167-1478013718598_facts_5_somehost.json
+
+Contains a version 5 "replace facts" command for certname "somehost"
+that was received at 1478013718598 milliseconds since the epoch
+(1970-01-01 UTC).
+
+Given this arrangement, you can use normal filesystem commands for
+rough examinations of the queue.  For example, something like this
+
+    find cmd/q | grep -cE '^cmd/q/[0-9]+-[0-9]+_facts_'
+
+should provide a count of "replace facts" commands in the queue, and
+something like this:
+
+    find cmd/q -printf "%s %p\n" | sort -n | tail
+
+should list the largest commands in the queue.
```

Comment:
Might be worth talking about how they're plain text and can be examined directly